src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -406,9 +411,28 @@ private OtpResponse getOtpResponse() {
     private OtpGraphQLVariables getQueryParamsForTargetZonedDateTime() {
         OtpGraphQLVariables params = trip.otp2QueryParams.clone();
         params.date = targetZonedDateTime.format(DateTimeUtils.DEFAULT_DATE_FORMATTER);
+        checkForRerouting(params);
         return params;
     }
 
+    /**
+     * Get the latest tracked journey associated with this trip, if available. If rerouting has occurred as part of that
+     * tracked journey, use the last rerouting location as the 'from place' instead of the original attributed to the
+     * trip.
+     */
+    public void checkForRerouting(OtpGraphQLVariables params) {
+        TrackedJourney trackedJourney = Persistence.trackedJourneys.getOneFiltered(
+            eq("tripId", trip.id),
+            Sorts.descending("dateCreated")
+        );
+        if (trackedJourney != null && trackedJourney.hasRerouted()) {
```

Comment:
Sorry, this check needs to include the condition that the journey has not ended. After the journey has ended, trip monitoring should resume with the original search params.

Reply:
Updated in: https://github.com/ibi-group/otp-middleware/pull/286/commits/03be35c3d73248ac38bc16909d206fec9569a2c7